chassis/invokers/src/main/java/com/griddynamics/jagger/invoker/http/v2/JHttpEndpoint.java

```diff
@@ -0,0 +1,139 @@
+package com.griddynamics.jagger.invoker.http.v2;
+
+import com.google.common.base.Preconditions;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static com.griddynamics.jagger.invoker.http.v2.JHttpEndpoint.Protocol.HTTP;
+import static com.griddynamics.jagger.invoker.http.v2.JHttpEndpoint.Protocol.HTTPS;
+import static java.lang.String.format;
+
+/**
+ * An object that represents HTTP-endpoint. It consists of {@link JHttpEndpoint#protocol},
+ * {@link JHttpEndpoint#hostname} and {@link JHttpEndpoint#port} fields. <p>
+ *
+ * @author Anton Antonenko
+ * @since 1.3
+ */
+@SuppressWarnings("unused")
+public class JHttpEndpoint {
+
+    /**
+     * Enum representing HTTP and HTTPS protocols
+     */
+    public enum Protocol {
+        HTTP("http"), HTTPS("https");
```

Comment:
passed argument just repeats constant name which can be easily extracted from enum